src/h5web/visualizations/nexus/utils.ts

```diff
@@ -71,3 +71,41 @@ export function getNxAxes(
 
   return { labels: axesLabels, ids: axesIds };
 }
+
+export function getNxDataGroup(
+  entity: HDF5Entity | undefined,
+  metadata: HDF5Metadata
+): HDF5Group | undefined {
+  if (!entity || !isGroup(entity)) {
+    return undefined;
+  }
+
+  if (isNxDataGroup(entity)) {
+    return entity;
+  }
+
+  const defaultPath = getAttributeValue(entity, 'default');
+
+  if (typeof defaultPath !== 'string') {
+    return undefined;
+  }
+
+  const relativePath = defaultPath.startsWith('/')
+    ? defaultPath.slice(1)
+    : defaultPath;
+
+  const defaultEntity = relativePath.split('/').reduce<HDF5Entity | undefined>(
+    (previousEntity, currentComponent) => {
+      if (!previousEntity) {
+        return undefined;
+      }
+      if (!isGroup(previousEntity)) {
+        return previousEntity;
+      }
```

Comment:
Don't you end up in a loop if the last entity in the path is a dataset? Shouldn't you return `undefined` in this case instead?

Reply:
Not that I know of. Let me break it down:
- the `reduce` will end up with `defaultEntity` being a dataset
- It will then call `getNxDataGroup(defaultEntity) 
- The guard clause at the beginning of the function `if(!entity | !isGroup(entity)` will make it return `undefined`. End of the story.

If I return `undefined`, the behaviour will be exactly the same except that `defaultEntity` will be `undefined` instead of a dataset.